test/index.js

```diff
@@ -121,51 +122,43 @@ describe('preBuild()', () => {
 })
 
 describe('onBuild()', () => {
-  test('runs next on netlify', async () => {
-    await plugin.onBuild({
-      constants: {
-        PUBLISH_DIR: '.',
-      },
-    })
-
-    expect(nextOnNetlify.mock.calls.length).toEqual(1)
-  })
-
-  test('calls makeDir with correct path', async () => {
+  test('runs NoN with functions_src & publish_dir options', async () => {
     const PUBLISH_DIR = 'some/path'
+    const FUNCTIONS_SRC = 'other/path'
     await plugin.onBuild({
       constants: {
         PUBLISH_DIR,
+        FUNCTIONS_SRC,
       },
     })
 
-    expect(await pathExists(PUBLISH_DIR)).toBeTruthy()
+    const nextOnNetlifyOptions = nextOnNetlify.mock.calls[0][0]
+    expect(nextOnNetlifyOptions.functionsDir).toEqual(FUNCTIONS_SRC)
+    expect(nextOnNetlifyOptions.publishDir).toEqual(PUBLISH_DIR)
```

Comment:
Sounds good :+1: 

Should we call the "real" `next-on-netlify` instead of mocking it? This would give a stronger confidence that the files are actually being copied to those directories. Generally speaking, this would be a stronger indicator that the plugin works as intended. What do you think? (Note: this would probably be better done in a separate PR)

Reply:
yes, definitely. when i tried to call the "real" nextOnNetlify at first, i got errors due to the emptyDirSync stuff in the prev NoN version. now we can address this in a separate PR. here's an issue https://github.com/netlify/netlify-plugin-nextjs/issues/53 :)

Reply:
Thanks for opening an issue for it! :heart: